python/idsse_common/test/test_http_utils.py

```diff
@@ -0,0 +1,161 @@
+    assert result[0] == EXAMPLE_FILES[0]
+
+
+def test_http_ls_on_error(http_utils: HttpUtils, httpserver: HTTPServer):
+    httpserver.expect_request('/data/'+EXAMPLE_ENDPOINT).respond_with_data('', content_type="text/plain")
+    result = http_utils.http_ls(EXAMPLE_URL + EXAMPLE_ENDPOINT)
+    assert result == []
+
+
+def test_http_cp_succeeds(http_utils: HttpUtils, httpserver: HTTPServer):
+    url = '/data/'+EXAMPLE_PROD_DIR+'/temp.grib2.gz'
+    httpserver.expect_request(url).respond_with_data(bytes([0,1,2]), status=200,
+                                                     content_type="application/octet-stream")
+    path = f'{EXAMPLE_URL}{EXAMPLE_PROD_DIR}/temp.grib2.gz'
+    dest = '/tmp/temp.grib2.gz'
+
+    copy_success = http_utils.http_cp(path, dest)
+    assert copy_success
+
+def test_http_cp_fails(http_utils: HttpUtils, httpserver: HTTPServer):
+    url = '/data/'+EXAMPLE_PROD_DIR+'/temp.grib2.gz'
+    httpserver.expect_request(url).respond_with_data(bytes([0, 1, 2]), status=404,
+                                                     content_type="application/octet-stream")
+    path = f'{EXAMPLE_URL}{EXAMPLE_PROD_DIR}/temp.grib2.gz'
+    dest = '/tmp/temp.grib2.gz'
+    copy_success = http_utils.http_cp(path, dest)
+    assert not copy_success
+
+
+def test_check_for_succeeds(http_utils: HttpUtils, httpserver: HTTPServer):
+    url = '/data/'+EXAMPLE_ENDPOINT
+    httpserver.expect_request(url).respond_with_data(EXAMPLE_RETURN, content_type="text/plain")
+
+    result = http_utils.check_for(EXAMPLE_ISSUE, EXAMPLE_VALID)
+    assert result is not None
+    assert result == (EXAMPLE_VALID, f'{EXAMPLE_URL}{EXAMPLE_PROD_DIR}{EXAMPLE_FILES[1]}')
+
+
+def test_check_for_does_not_find_valid(http_utils: HttpUtils, httpserver: HTTPServer):
+    url = '/data/'+EXAMPLE_ENDPOINT
+    httpserver.expect_request(url).respond_with_data('', content_type="text/plain")
+    unexpected_valid = datetime(1970, 10, 3, 23, tzinfo=UTC)
+    result = http_utils.check_for(EXAMPLE_ISSUE, unexpected_valid)
+    assert result is None
+
+
+def test_get_issues(http_utils: HttpUtils, httpserver: HTTPServer):
+    url = '/data/' + EXAMPLE_ENDPOINT +'/'
+    httpserver.expect_request(url).respond_with_data(EXAMPLE_RETURN, content_type="text/plain")
+    result = http_utils.get_issues(issue_start=EXAMPLE_ISSUE,
+                                   time_delta=timedelta(minutes=1))
+    assert len(result) == 1
+    assert result[0] == EXAMPLE_ISSUE
+
+
+def test_get_issues_with_same_start_stop(http_utils: HttpUtils, httpserver: HTTPServer):
+    url = '/data/'+EXAMPLE_ENDPOINT+'/'
+    httpserver.expect_request(url).respond_with_data(EXAMPLE_RETURN, content_type="text/plain")
+    result = http_utils.get_issues(issue_start=EXAMPLE_ISSUE, issue_end=EXAMPLE_ISSUE, time_delta=timedelta(minutes=1))
+    assert len(result) == 1
+    assert result[0] == EXAMPLE_ISSUE
+
+def test_get_valids_all(http_utils: HttpUtils, httpserver: HTTPServer):
```

Comment:
Given that this http_utils.get_valids doesn't find anything, is this a meaningful test?

Reply:
This does return a list of files based on the pattern provided so I feel that this is handled.

Reply:
This return an empty list because the sample filename you used don't have and valid info. If the files were structured list this:
'MRMS_MergedReflectivityQC_00.50_{issue.year:04d}{issue.month:02d}{issue.day:02d}'
'-{issue.hour:02d}{issue.minute:02d}{issue.second:02d}'_{valid.year:04d}{valid.month:02d}'
'{valid.day:02d}-{valid.hour:02d}{valid.minute:02d}{valid.second:02d}.grib2.gz'
It should return a valid datetime that matches the issue. We want http_utils to be generic enough to handle more than MRMS so we need to test that it can get valids.